src/weatherSource/WeatherSource.js

```diff
@@ -1,129 +1,90 @@
+
+            } else if (params.city) {
+                constructUrl = `q=${params.city}`
+
+            } else {
+                throw ('Wrong parameters')
+            }
 
         } else {
             throw 'No parameters were given'
+
         }
-        return constructUrl;
+        return constructUrl + `&APPID=${this.API_KEY}&units=metric`;
     }
-    // zwraca obiekt 1pogodowy
-    // w parametrach należy podać dwie wartości string:
-    // pierwszą 'miasto' i drugą 'państwo'
-    static async getWeatherByPlace(city, country) {
-
-        let dataUnparsed;
-
-        country
-            ?
-            dataUnparsed = await fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city},${country}&APPID=${API_KEY}&units=metric`) :
-            dataUnparsed = await fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city}&APPID=${API_KEY}&units=metric`)
-
-        if (dataUnparsed.ok) {
-            let data = await dataUnparsed.json();
-            // wyświetlenie całej zawartości odpowiedzi 
-            // console.log(data);
-
-            const dataModel = {
-                main: data.main,
-                clouds: data.clouds,
-                weather: data.weather,
-                wind: data.wind,
-                place: data.name,
-                coords: data.coord
-            }
 
-            return dataModel
+    // metoda prywatna zwracająca potrzebne dane o 
+    // obecnej pogodzie z otrzymanego obiektu API
+    _constructCurrentWeatherObject(data) {
+        return {
+            main: data.main,
+            clouds: data.clouds,
+            weather: data.weather,
+            wind: data.wind,
+            place: data.name,
+            coords: data.coord
         }
-
-        throw dataUnparsed;
     }
 
-    // zwraca obiekt pogodowy
-    // W parametrach należy przekazać obiekt z polami:
-    // latitude(szerokośc geo.) i longitude(długość geo.) 
-    static async getWeatherByCoordinates(coords) {
-
-        let dataUnparsed = await fetch(`http://api.openweathermap.org/data/2.5/weather?lat=${coords.latitude}&lon=${coords.longitude}&APPID=${API_KEY}`)
+    //TODO zrobić model prognozy
+    _constructForecastWeatherObject(data) {
+        return data;
+    }
+    
+    // metoda prywatna odpowiadająca za główną mechanikę pobierania danych z Api
+    async _getWeather(params, weatherType, doo) {
+        let constructUrl = this._verify(params)
+        
+        let dataUnparsed = await fetch(this.API_URL_ADDRESS + weatherType + '?' + constructUrl)
         if (dataUnparsed.ok) {
             let data = await dataUnparsed.json();
-            // console.log(data);
-
-            const dataModel = {
-                main: data.main,
-                clouds: data.clouds,
-                weather: data.weather,
-                wind: data.wind,
-                place: data.name,
-                coords: data.coord
-            }
-
-            return dataModel
+            return doo(data)
```

Comment:
Musiałem sobie to wszystko ściągnąć, żeby znaleźć, co to jest doo, więc przypuszczam, że dałoby się lepiej ten parametr nazwać..

Reply:
Aaa tak tak. To już moja pomyłka. Nie miałem pomysłu na nazwanie tej zmiennej więc zostawiłem ją tak. Miałem ja potem zmienić, aż w końcu zapomniałem